### ISSUE TITLE
Publishing binary payload in gRPC pubsub results in JS SDK not setting proper content type

### ISSUE THREAD

**Issue by alecor191** [dapr/js-sdk#448]
## Expected Behavior

When providing a `Buffer` object to `pubsub.publish()`, the content type should be set to `application/octet-stream` (reflected in `datacontenttype` of the CloudEvent).

## Actual Behavior

Content type is set to `text/plain` no matter what content is sent.

## Steps to Reproduce the Problem

Call `pubsub.publish()` with a `Buffer` as payload.

## Observations in GRPCClient's `pubsub.js`

I noticed that when publishing a message with the JS SDK, function `serializeGrpc()` in `Serializer.util.js` will be called. This function looks as follows:

```javascript
function serializeGrpc(data) {
    let serializedData = data;
    let contentType = "application/octet-stream";
    if (!(data instanceof Buffer)) {
        serializedData = Buffer.from(JSON.stringify(data), "utf-8");
        contentType = "application/json";
    }
    return { serializedData, contentType };
}
```

As I want to send some binary data, I made sure to pass a `Buffer` to `publish()`, so that JS SDK recognizes it as binary data.

Unfortunately, the returned `contentType` is not used by the calling `GRPCClientPubSub.publish()` function. Instead, it only uses the `serializedData` property. Here the relevant code snippet taken from that function:

```javascript
msgService.setData(SerializerUtil.serializeGrpc(data).serializedData);
```

Could it be that in this code in `GRPCClientPubSub.publish()`:

```javascript
const msgService = new dapr_pb_1.PublishEventRequest();
msgService.setPubsubName(pubSubName);
msgService.setTopic(topic);
msgService.setData(SerializerUtil.serializeGrpc(data).serializedData);
```

a call to `msgService.setDataContentType()` is missing, setting it to the value of `contentType` property returned by `SerializerUtil.serializeGrpc(data)`?

Also note that with above logic, content type is also never set to `application/json`, even though `serializeGrpc()` returns that content type for JSON payload.

**Comment by XavierGeerinck** [comment]
Hi there!

thank you for opening this. We are indeed aware of issues with the serialization and have added changes coming in 3.0.0 that will address this for both HTTP and gRPC

we hope to have it released as soon as possible and can’t wait to bring all the new things and fixes to you 😊

kind regards,
Xavier Geerinck

**Comment by XavierGeerinck** [comment]
FYI more work was done on this in PR https://github.com/dapr/js-sdk/pull/453 

In the current pubsub implementation, the content type is being set (see below). Tests are also in the upcoming PR to remediate more issues regarding serialization to tackle this earlier

```
    if (data) {
      const serialized = SerializerUtil.serializeGrpc(data);
      msgService.setData(serialized.serializedData);
      msgService.setDataContentType(serialized.contentType);
    }
```

**Comment by XavierGeerinck** [comment]
Will be closed in the upcoming 3.0.0 release (target this friday)
